packages/lit-html/src/lib/lit-html.ts

```diff
@@ -189,8 +189,16 @@ class Template {
     // we also dynamically create a regex to find the matching end tags for raw
     // text elements.
 
-    const textRegex = /<(!--|\/?\w+|(\W|$))/g;
+    // TODO (justinfagnani): we detect many more parsing edge-cases than we
+    // used to, and many of those are of dubious value. Decide and document
+    // how to relax correctness to simplify the regexes and states.
+
+    // End of text is: `<` followed by:
+    // (comment start) or (tag) or (unescaped <) or (dynamic tag)
+    const textRegex = /<(?:(!--|\/[^a-zA-Z])|(\/?[a-zA-Z]\w*)|(\s)|(\/?$))/g;
```

Comment:
What's the whitespace for? I think this will break the search for `</script>` because we're using the full `match[2]` to create the dynamic regex. Eg, in `<script >foo</script> this text should be outside the script`

```suggestion
    const textRegex = /<(?:(!--|\/[^a-zA-Z])|(\/?[a-zA-Z])|(\s)|(\/?$))/g;
```

Reply:
That's a word char, not whitespace. This did remind me to expand those characters to include support for `-`, `:`, etc in tag names. Though this won't matter until the attribute support is added.

Reply:
Lol. 😳